src/t_list.c

```diff
@@ -441,19 +650,23 @@ void addListRangeReply(client *c, robj *o, long start, long end, int reverse) {
 
     /* Return the result in form of a multi-bulk reply */
     addReplyArrayLen(c,rangelen);
-    if (o->encoding == OBJ_ENCODING_QUICKLIST) {
+    if (o->encoding == OBJ_ENCODING_QUICKLIST || o->encoding == OBJ_ENCODING_LISTPACK) {
```

Comment:
This `if` is unnecessary. The body of this uses generic listType functions that don't depend on encoding.

Reply:
Done